Source/Renderer/modernizeShader.js

```diff
@@ -0,0 +1,233 @@
+                            });
+                        }
+                    }
+                }
+            }
+        }
+
+        for (var care in variableMap) {
+            if (variableMap.hasOwnProperty(care)) {
+                if (variableMap.length === 1 && variableMap[0] === null) {
+                    variableMap.splice(0, 1);
+                }
+            }
+        }
+
+        return variableMap;
+    }
+
+    function removeExtension(name, splitSource) {
+        var regex = '#extension\\s+GL_' + name + '\\s+:\\s+[a-zA-Z0-9]+\\s*$';
+        replaceInSourceRegex(new RegExp(regex, 'g'), '', splitSource);
+    }
+
+    /**
+     * A function to port GLSL shaders from GLSL ES 1.00 to GLSL ES 3.00
+     *
+     * This function is nowhere near comprehensive or complete. It just
+     * handles some common cases.
+     *
+     * Note that this function requires the presence of the
+     * "#define OUTPUT_DECLARATION" line that is appended
+     * by ShaderSource.
+     *
+     * @private
+     */
+    function modernizeShader(source, isFragmentShader) {
+        var outputDeclarationRegex = /#define OUTPUT_DECLARATION/;
+        var splitSource = source.split('\n');
+
+        if (/#version 300 es/g.test(source)) {
+            return source;
+        }
+
+        var outputDeclarationLine = -1;
+        var number;
+        for (number = 0; number < splitSource.length; ++number) {
+            var line = splitSource[number];
+            if (outputDeclarationRegex.exec(line)) {
+                outputDeclarationLine = number;
+                break;
+            }
+        }
+
+        if (outputDeclarationLine === -1) {
+            throw new DeveloperError('Could not find a #define OUTPUT_DECLARATION!');
+        }
+
+        var outputVariables = [];
+
+        for (var i = 0; i < 10; i++) {
+            var fragDataString = 'gl_FragData\\[' + i + '\\]';
+            var newOutput = 'czm_out' + i;
+            var regex = new RegExp(fragDataString, 'g');
+            if (regex.test(source)) {
+                setAdd(newOutput, outputVariables);
+                replaceInSourceString(fragDataString, newOutput, splitSource);
+                splitSource.splice(outputDeclarationLine, 0, 'layout(location = ' + i + ') out vec4 ' + newOutput + ';');
+                outputDeclarationLine += 1;
+            }
+        }
+
+        var czmFragColor = 'czm_fragColor';
+        if (findInSource('gl_FragColor', splitSource)) {
+            setAdd(czmFragColor, outputVariables);
+            replaceInSourceString('gl_FragColor', czmFragColor, splitSource);
+            splitSource.splice(outputDeclarationLine, 0, 'layout(location = 0) out vec4 czm_fragColor;');
+            outputDeclarationLine += 1;
+        }
+
+        var variableMap = getVariablePreprocessorBranch(outputVariables, splitSource);
+        var lineAdds = {};
+        for (var c = 0; c < splitSource.length; c++) {
```

Comment:
Same comment. For consistency just use `i` throughout and declare `var i;` before it is first used.